src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,74 @@
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final RequestDtoMapper<ProductRequestDto, Product>
+            productRequestDtoMapper;
+    private final ResponseDtoMapper<ProductResponseDto, Product>
+            productResponseDtoMapper;
+
+    public ProductController(ProductService productService,
+                             RequestDtoMapper<ProductRequestDto, Product> productRequestDtoMapper,
+                             ResponseDtoMapper<ProductResponseDto, Product>
+                                     productResponseDtoMapper) {
+        this.productService = productService;
+        this.productRequestDtoMapper = productRequestDtoMapper;
+        this.productResponseDtoMapper = productResponseDtoMapper;
+    }
+
+    @PostMapping
+    public ProductResponseDto save(@RequestBody ProductRequestDto productRequestDto) {
+        Product product = productService.save(productRequestDtoMapper
+                .mapToModel(productRequestDto));
+        productService.save(product);
+        return productResponseDtoMapper.mapToDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto findById(@PathVariable Long id) {
+        return productResponseDtoMapper.mapToDto(productService.findById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteById(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto productRequestDto) {
+        Product product = productRequestDtoMapper.mapToModel(productRequestDto);
+        product.setId(id);
+        productService.save(product);
+        return productResponseDtoMapper.mapToDto(product);
+    }
+
+    @GetMapping("/price")
+    public List<ProductResponseDto> findAllByPriceBetween(@RequestParam BigDecimal from,
+                                                          @RequestParam BigDecimal to) {
+        return productService.findAllByPriceBetween(from, to).stream()
+                .map(productResponseDtoMapper::mapToDto)
+                .collect(Collectors.toList());
+    }

```

Comment:
Don't see this method
![Screenshot 2023-03-16 at 11 09 15](https://user-images.githubusercontent.com/60258373/225568781-bafcac1c-ae86-41a8-ab46-e44b5c2dba9c.png)